sdk/go/pulumi/stack_reference.go

```diff
@@ -13,14 +15,26 @@ type StackReference struct {
 	Name StringOutput `pulumi:"name"`
 	// Outputs resolves with exports from the named stack
 	Outputs MapOutput `pulumi:"outputs"`
+
+	// Engine context in which resolve resource references in outputs.
+	Context *Context
```

Comment:
I was reluctant to add this, as I'm unsure of the mechanics here. Can stacks export a resource as an output, and that would _work_?

`unmarshalPropertyValue` requires a valid Pulumi engine Context to handle that case. Is it worthwhile to split that function to support this use case?